modules/base/shaders/billboardpoint_fs.glsl

```diff
@@ -24,45 +24,86 @@
 
 #include "fragment.glsl"
 
-flat in vec4 gs_colorMap;
+flat in float gs_colorParameter;
 flat in float vs_screenSpaceDepth;
 in vec2 texCoord;
-in float ta;
 
 uniform float alphaValue;
 uniform vec3 color;
+
+uniform vec4 nanColor = vec4(0.5);
+uniform bool useNanColor = true;
+
+uniform vec4 aboveRangeColor;
+uniform bool useAboveRangeColor;
+
+uniform vec4 belowRangeColor;
+uniform bool useBelowRangeColor;
+
+uniform bool hasSpriteTexture;
 uniform sampler2D spriteTexture;
-uniform bool hasColorMap;
+
 uniform bool useColorMap;
+uniform sampler1D colorMapTexture;
+uniform float cmapRangeMin;
+uniform float cmapRangeMax;
+uniform bool hideOutsideRange;
+
 uniform float fadeInValue;
 
+vec4 sampleColorMap(float dataValue) {
```

Comment:
This is the shader snippet that the ColorMappingComponent could generate. 

How best to do this?